scene/gui/tab_container.cpp

```diff
@@ -479,7 +481,14 @@ void TabContainer::set_current_tab(int p_current) {
 	}
 
 	_change_notify("current_tab");
-	emit_signal("tab_changed",current);
+
+	if (pending_previous == current)
+		emit_signal("tab_selected", current);
```

Comment:
What I mean is that `tab_selected` should be emitted when you select a `!current` tab, otherwise it doesn't make any sense and should be named `current_tab_reselected`.

You should not delete your fork when you have pending PRs...